src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5815,6 +5815,20 @@ private static void CombineExtensionMethodArguments(BoundExpression receiver, An
                 HashSet<DiagnosticInfo> useSiteDiagnostics = null;
                 OverloadResolution.MethodInvocationOverloadResolution(methodGroup.Methods, methodGroup.TypeArguments, actualArguments, overloadResolutionResult, ref useSiteDiagnostics, isMethodGroupConversion, allowRefOmittedArguments);
                 diagnostics.Add(expression, useSiteDiagnostics);
+
+                if (overloadResolutionResult.Succeeded)
+                {
+                    var receiverParameter = overloadResolutionResult.BestResult.Member.Parameters.FirstOrDefault();
+                    Debug.Assert(receiverParameter != null);
```

Comment:
>Debug.Assert(receiverParameter != null); [](start = 20, length = 40)

Please add cast to object. Also, why not use First instead of FirstOrDefault then?